pubtools/tracing.py

```diff
@@ -0,0 +1,123 @@
+"""Instrument Pub task functions.
+
+Usage:
+    @instrument_func()
+      def func():
+          pass
+
+"""
+import functools
+import logging
+import os
+import threading
+
+from opentelemetry import baggage, context, trace
+from opentelemetry.baggage.propagation import W3CBaggagePropagator
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import \
+    OTLPSpanExporter
+from opentelemetry.propagate import set_global_textmap
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.trace import Status, StatusCode
+from opentelemetry.trace.propagation.tracecontext import \
+    TraceContextTextMapPropagator
+
+propagator = TraceContextTextMapPropagator()
+baggage_propagator = W3CBaggagePropagator()
+log = logging.getLogger(__name__)
+
+
+class TracingWrapper:
```

Comment:
There's no delineation here between what's intended as public API, and what's considered just an implementation detail.

As far as I can tell, only the `instrument_func` decorator should be public API.  Everything else should be hidden.

Can you please follow the same setup as other modules in order to achieve this?  Check how pubtools/pluggy.py works for example. Basically it means that:

- this file should be moved to `pubtools/_impl/tracing.py` and it can contain anything you want
- in `pubtools/tracing.py` you should import only what's intended as public API - `instrument_func` I guess - and put it into `__all__` .

Reply:
Updated as your suggestion.